### PR TITLE
Update documentation for pattern matching cache delete

### PR DESCRIPTION
**Pull request by calvincestari**
This updates the [documentation for cache delete](https://www.apollographql.com/docs/ios/caching/#delete) - _it should have been part of #1848_.

#### How to test the changes
1. Navigate to `apollo-ios/docs` in a terminal window
2. Execute `npm install`
3. Execute `npm start`
4. Open a browser to the link provided in the console

- [ ] Validate that the documentation for 'Direct cache access' -> 'delete' shows the updates and makes sense.

### CODE REVIEW

**Comment by AnthonyMDev** [comment]
We should wait to push this until we are ready to cut the next release. Docs get deployed as soon as they are pushed to `main`.

File: docs/source/caching.mdx [hunk 1]
```diff
@@ -180,14 +180,15 @@ store.withinReadWriteTransaction({ transaction in
 
 ### delete
 
-Delete functionality is limited at this time. There are presently two deletion methods available on `ApolloStore`, which are supported by both the in memory and the `SQLite` caches:
+Delete functionality is limited at this time. There are presently three deletion methods available on `ApolloStore`, which are supported by both the in memory and the `SQLite` caches:
 
 1. `clear` - Removes everything from the cache immediately. This is basically the "Nuke it from orbit, it's the only way to be sure" option. 
 2. `removeObject(for:)` on `ReadWriteTransaction`. Removes a single object for the given `CacheKey`. 
+3. `removeObjects(matching:)` on `ReadWriteTransaction`. Removes all objects with a `CacheKey` that matches the given pattern. Pattern matching is **not** case sensitive. For an in memory cache it is the equivalent of checking whether the cache key _contains_ the pattern and `SQLite` caches will perform a `LIKE` query to remove the objects.
```
Comment (designatednerd) [inline docs/source/caching.mdx:187]: Probably worth noting that this is slow as hell because it goes through every cache key in the whole cache to work

**Review by designatednerd — APPROVED** [review]
one note, otherwise LGTM

**Comment by designatednerd** [comment]
:shipit:

**Comment by AnthonyMDev** [comment]
I'm merging this into the `networking-docs-update` branch, which is poorly named now, as I'm just using it as the branch for the 0.45.0 release.

**Pull request merged**